### PR TITLE
Map article search series to catalog search series

### LINKED ISSUE #1588: Map "Series" to "Series' in catalog/article switch
From the original ticket (#1489)  :

drh: Also shouldn't "Series" (Catalog) map to "Journal title (source)"?

jv: No. I don't think this would ever get a relevant result. The MARC Series title field is not going to contain journal titles; and EDS has eBook series that will match in SE.

These are equivalent searches:

https://sw-webapp-sandbox-f.stanford.edu/catalog?utf8=%E2%9C%93&op=AND&search=&search_title=&search_author=&subject_terms=&series_search=SpringerBriefs+in+Environmental+Science&pub_search=&isbn_search=&range%5Bpub_year_tisim%5D%5Bbegin%5D=&range%5Bpub_year_tisim%5D%5Bend%5D=&sort=relevance&search_field=advanced&button=
https://sw-webapp-sandbox-f.stanford.edu/articles?utf8=%E2%9C%93&search_field=search&q=SE+%22SpringerBriefs+in+Environmental+Science%22

This is currently implemented as Series -> Journal title - needs to be updated to Series -> Series.

### PR DESCRIPTION
**Pull request by camillevilla**
Closes #1588 

This PR updates catalog --> article search parameter mapping. Catalog `search_series` now maps to article `series` (field code `SE`). 

## Example searches:
- [Catalog search](http://localhost:3000/?q=SpringerBriefs+in+Environmental+Science&search_field=search_series)
- [Articles search](http://localhost:3000/articles?q=SpringerBriefs+in+Environmental+Science&search_field=series)

## Demo
series-mapping

### CODE REVIEW

File: app/controllers/articles_controller.rb [hunk 1]
```diff
@@ -72,6 +73,10 @@ def eds_init
       field.label = 'Abstract'
     end
 
+    config.add_search_field('series') do |field|
+      field.label = 'Series'
```
Comment (jkeck) [inline app/controllers/articles_controller.rb]: Do we know if series is a supported search type from the EDS gem?  Looking over the [code in master](https://github.com/ebsco/edsapi-ruby/blob/71eddcda215cbcf4455dc5758d04b86ff9406af9/lib/ebsco/eds/options.rb#L344-L368), it doesn't seem like it is?

Comment (camillevilla) [inline app/controllers/articles_controller.rb]: @jkeck interesting! I just noticed @jvine's example search uses the field code `SE` and returns 5 results. My current query returns 59. So the search is supported, it's just not configured as a search field. Should I try to make a PR on the EBSCO repo?

Comment (camillevilla) [inline app/controllers/articles_controller.rb]: Examples article search: http://localhost:3000/articles?utf8=%E2%9C%93&search_field=search&q=SE+%22SpringerBriefs+in+Environmental+Science%22

**Comment by camillevilla** [comment]
alrighty @jkeck! Upgrading to `ebsco-eds 0.3.3` is done and this PR's ready for review.

**Pull request merged**